WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -278,6 +278,20 @@ def update_events
     end
   end
 
+  def update_wcif_events
+    @competition = competition_from_params
+    wcif_events = params["_json"].map { |wcif_event| wcif_event.permit!.to_h } #<<< TODO: validate that the incoming stuff is actually wcif >>>
```

Comment:
1. Yeah, I also see no obvious way to attack this, and there probably isn't one, but that doesn't feel paranoid enough.

2. If I had my way, every web request should be one DB transaction. This solves a lot of paranoid problems. Easily done in an around filter.

3. Makes sense.